test/Tests/src/System/File_Spec.enso

```diff
@@ -15,32 +19,18 @@ spec =
             file = File.new sample_file
             file . should_equal sample_file
 
-        Test.specify "should allow reading a file" <|
-            contents = sample_file.read
-            contents.should .start_with "Cupcake ipsum dolor sit amet."
-
-        Test.specify "should allow reading a file to byte vector" <|
-            contents = sample_file.read_bytes
-            contents.take_start 6 . should_equal [67, 117, 112, 99, 97, 107]
-
-        Test.specify "should handle exceptions when reading a non-existent file" <|
-            file = File.new "does_not_exist.txt"
-            successfully_failed = Panic.recover Any file.read . catch e-> case e of
-                File.No_Such_File_Error _ -> True
-                _ -> False
-            successfully_failed . should_be_true
-
         Test.specify "should check if file exists" <|
             non_existent_file.exists.should_be_false
             sample_file.exists.should_be_true
 
-        Test.specify "should check if file is a directory" <|
-            sample_file.is_directory.should_be_false
-            Enso_Project.root.is_directory.should_be_true
-
         Test.specify "should get file name" <|
             sample_file.name.should_equal "sample.txt"
 
+        Test.specify "should get file extension" <|
+            sample_file.extension.should_equal ".txt"
+            (File.new "this.is.a.silly.name.txt").extension.should_equal ".txt"
+            (File.new "no_extension").extension.should_equal ""
```

Comment:
I'm wondering if returning `Nothing` instead of `""` wouldn't be more Enso-style?
Also wondering what should be done for filename `foo.` - I'd suggest adding this test case even if only for documentation purposes (but also a good edge case for testing any off-by-one errors).